Disable the clang-format checks when building DXC

**What is the issue you're encountering?**
When I build the `dxc` target from source as a subdirectory of my codebase, I get these errors:

> 3>Verifying clang-format results...
3>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'build\win\CMakeFiles\88778fe0101fa821569a6b8d53ded241\RDAT_LibraryTypes.inl.stamp.rule;build\win\CMakeFiles\6f5c683015713da5c9349b7bc8142ac1\RDAT_LibraryTypes.rule' exited with code 1.

A few more targets also fail with this same error. It still produces `dxc.exe` and `dxcompiler.dll`. Is it possible to turn off the clang-format checks? 

**Steps to Reproduce (if applicable)**
1. Clone the latest master
2. Build the dxc target as a subdirectory of another CMake project


**Environment**
- DXC version: `dxcompiler.dll: 1.8 - 1.8.0.2207 (master, 452045c6f-dirty)`
- Host Operating System: Windows 10 22H2 1904.4529
- CMake: `cmake version 3.28.3-msvc11`
- Compiler: `MSVC 19.40.33811.0`

--- a/cmake/modules/HCT.cmake
+++ b/cmake/modules/HCT.cmake
@@ -1,4 +1,6 @@
 option(HLSL_COPY_GENERATED_SOURCES "Copy generated sources if different" Off)
+option(HLSL_DISABLE_SOURCE_GENERATION "Disable generation of in-tree sources" Off)
+mark_as_advanced(HLSL_DISABLE_SOURCE_GENERATION)
 
 add_custom_target(HCTGen)
 
@@ -42,6 +44,10 @@ function(add_hlsl_hctgen mode)
 
   if (NOT ARG_OUTPUT)
     message(FATAL_ERROR "add_hlsl_hctgen requires OUTPUT argument")
+  endif()
+
+  if (HLSL_DISABLE_SOURCE_GENERATION AND NOT ARG_BUILD_DIR)
+    return()
   endif()
  
   set(temp_output ${CMAKE_CURRENT_BINARY_DIR}/tmp/${ARG_OUTPUT})
@@ -101,7 +107,7 @@ function(add_hlsl_hctgen mode)
   endif()
 
   add_custom_command(OUTPUT ${temp_output}
-                     COMMAND ${Python3_EXECUTABLE}
+                     COMMAND ${PYTHON_EXECUTABLE}
                              ${hctgen} ${force_lf}
                              ${mode} --output ${temp_output} ${input_flag}
                      ${format_cmd}

--- a/cmake/modules/HCT.cmake
+++ b/cmake/modules/HCT.cmake
@@ -107,7 +107,7 @@ function(add_hlsl_hctgen mode)
   endif()
 
   add_custom_command(OUTPUT ${temp_output}
-                     COMMAND ${PYTHON_EXECUTABLE}
+                     COMMAND ${Python3_EXECUTABLE}
                              ${hctgen} ${force_lf}
                              ${mode} --output ${temp_output} ${input_flag}
                      ${format_cmd}

